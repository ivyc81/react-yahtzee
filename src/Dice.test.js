import React from 'react';
import { shallow, mount } from "enzyme";
import Dice from './Dice';
import toJson from "enzyme-to-json";

const DICE = [1,2,3,4,5];
const LOCKED = [false, false, false, true, true];

it('renders without crashing', () => {
  shallow(<Dice dice={DICE} locked={ LOCKED }/>);
});

it('renders correctly ', () => {

    const wrapper = mount(<Dice dice={DICE} locked={ LOCKED }/>);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});
