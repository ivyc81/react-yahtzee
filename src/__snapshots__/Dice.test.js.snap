// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly  1`] = `
<Dice
  dice={
    Array [
      1,
      2,
      3,
      4,
      5,
    ]
  }
  locked={
    Array [
      false,
      false,
      false,
      true,
      true,
    ]
  }
>
  <div
    className="Dice"
  >
    <Die
      idx={0}
      key="0"
      locked={false}
      val={1}
    >
      <button
        className="Die"
        onClick={[Function]}
      >
        1
      </button>
    </Die>
    <Die
      idx={1}
      key="1"
      locked={false}
      val={2}
    >
      <button
        className="Die"
        onClick={[Function]}
      >
        2
      </button>
    </Die>
    <Die
      idx={2}
      key="2"
      locked={false}
      val={3}
    >
      <button
        className="Die"
        onClick={[Function]}
      >
        3
      </button>
    </Die>
    <Die
      idx={3}
      key="3"
      locked={true}
      val={4}
    >
      <button
        className="Die Die-locked"
        onClick={[Function]}
      >
        4
      </button>
    </Die>
    <Die
      idx={4}
      key="4"
      locked={true}
      val={5}
    >
      <button
        className="Die Die-locked"
        onClick={[Function]}
      >
        5
      </button>
    </Die>
  </div>
</Dice>
`;
